cucumber/authenticators_gcp/features/step_definitions/authn_gcp_steps.rb

```diff
@@ -0,0 +1,31 @@
+Given(/^I set (authn-gcp\/service-account-id|authn-gcp\/service-account-email|authn-gcp\/project-id|authn-gcp\/instance-name) annotation (with incorrect value )?to host "([^"]*)"$/) do |annotation_name, incorrect_value, hostname|
+  i_have_a_resource "host", hostname
+  case annotation_name
+  when "authn-gcp/service-account-id"
+    annotation_value = gcp_service_account_id
+  when "authn-gcp/service-account-email"
+    annotation_value = gcp_service_account_email
+  when "authn-gcp/project-id"
+    annotation_value = gcp_project_id
+  when "authn-gcp/instance-name"
+    annotation_value = gce_instance_name
+  else
+    raise "incorrect annotation name #{annotation_name}"
+  end
+
+  set_annotation_to_resource("#{annotation_name}", annotation_value)
+end
+
+Given(/^I obtain a GCE identity token in "([^"]*)" format with audience claim value: "([^"]*)"$/) do | audience, format |
+  gce_identity_access_token(
+    audience: audience,
+    token_format: format
+  )
+end
+
+Given(/I authenticate with authn-gcp using GCP identity token as host "([^"]*)"/) do |username|
```

Comment:
you are not using the username. is this on purpose?

Reply:
again merge